### PR TITLE
fix: exclude supabase managed schemas from inspect subcommand

### PR DESCRIPTION
**Pull request by encima**
## What kind of change does this PR introduce?

Feature/Fix: moving queries for inspect to the root `inspect` module and removing Supabase schemas from queries

## What is the current behavior?

Queries are within the command and show results from Supabase schemas that could cause concern or make the user take actions on schemas they do not own

## What is the new behavior?

- Queries moved into a single place (could be further developed to template similar queries)
- Supabase schemas removed from relevant queries
- Explanation of hit rates added to `cache-hit` command and a column explains whether action might be needed

## Additional context

image

### CODE REVIEW

File: internal/inspect/table_index_sizes/table_index_sizes.go [hunk 1]
```diff
@@ -18,7 +18,7 @@ SELECT c.relname AS table,
   pg_size_pretty(pg_indexes_size(c.oid)) AS index_size
 FROM pg_class c
 LEFT JOIN pg_namespace n ON (n.oid = c.relnamespace)
-WHERE n.nspname NOT IN ('pg_catalog', 'information_schema')
+WHERE n.nspname NOT IN ('pg_catalog', 'information_schema', 'auth', 'realtime', 'pgsodium', 'supabase_migrations', 'vault', 'storage')
```
Comment (sweatybridge) [inline internal/inspect/table_index_sizes/table_index_sizes.go:21]: Why are we moving some queries but not this one?

Can we also split the refactor into a separate commit so it's easier to review what's changed?

There's a full list of supabase managed schemas defined in misc.go. I think we should use it instead.

Comment (encima) [reply internal/inspect/table_index_sizes/table_index_sizes.go:21]: missed the removal of that query 🙏

File: internal/inspect/queries.go [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+SELECT
+  'index' as type,
+  schemaname,
+  tablename || '::' || iname as object_name,
+  ROUND(CASE WHEN iotta=0 OR ipages=0 THEN 0.0 ELSE ipages/iotta::numeric END,1) AS bloat,
+  CASE WHEN ipages < iotta THEN '0' ELSE (bs*(ipages-iotta))::bigint END AS raw_waste
+FROM
+  index_bloat) bloat_summary
+WHERE
+ schemaname NOT IN (%s)
```
Comment (sweatybridge) [inline internal/inspect/queries.go:82]:
```suggestion
 schemaname NOT IN ($1)
```

I would use a placeholder here and pass in an array of strings to pgx client, for eg. 
 `conn.Query(sql, excluded)`. It's more robust than golang format strings and will probably simplify the code a little bit.

**Review by sweatybridge — COMMENTED** [review]
Just one more comment. Everything else looks good.

File: internal/inspect/table_sizes/table_sizes.go [hunk 3]
```diff
@@ -18,7 +18,7 @@ SELECT c.relname AS name,
   pg_size_pretty(pg_table_size(c.oid)) AS size
 FROM pg_class c
 LEFT JOIN pg_namespace n ON (n.oid = c.relnamespace)
-WHERE n.nspname NOT IN ('pg_catalog', 'information_schema')
+WHERE n.nspname NOT IN ('pg_catalog', 'information_schema', 'auth', 'realtime', 'pgsodium', 'supabase_migrations', 'vault', 'storage')
```
Comment (sweatybridge) [inline internal/inspect/table_sizes/table_sizes.go:21]: We missed out this one too.

File: internal/inspect/queries.go [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+
+const CALLS_QUERY = `SELECT
+'index hit rate' AS name,
+(sum(idx_blks_hit)) / nullif(sum(idx_blks_hit + idx_blks_read),0) AS ratio
+FROM pg_statio_user_indexes
+UNION ALL
+SELECT
+'table hit rate' AS name,
+sum(heap_blks_hit) / nullif(sum(heap_blks_hit) + sum(heap_blks_read),0) AS ratio
+FROM pg_statio_user_tables`
```
Comment (sweatybridge) [inline internal/inspect/queries.go:102]:
```suggestion
const CALLS_QUERY = `SELECT
	query,
	interval '1 millisecond' * total_exec_time AS total_exec_time,
	to_char((total_exec_time/sum(total_exec_time) OVER()) * 100, 'FM90D0') || '%'  AS prop_exec_time,
	to_char(calls, 'FM999G999G990') AS ncalls,
	interval '1 millisecond' * (blk_read_time + blk_write_time) AS sync_io_time
FROM pg_stat_statements	
ORDER BY calls DESC
LIMIT 10`
```

File: internal/inspect/table_sizes/table_sizes.go (hunk 3 above)
Comment (encima) [reply internal/inspect/table_sizes/table_sizes.go]: that was probably my poor rebasing! fixed now!

Comment (encima) [reply internal/inspect/table_sizes/table_sizes.go]: see that you fixed this already, removed my commit

File: internal/inspect/queries.go [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+ORDER BY calls DESC
+LIMIT 10`
+
+const INDEX_SIZE_QUERY = `SELECT c.relname AS name,
+pg_size_pretty(sum(c.relpages::bigint*8192)::bigint) AS size
+FROM pg_class c
+LEFT JOIN pg_namespace n ON (n.oid = c.relnamespace)
+WHERE n.nspname NOT IN (%1)
+AND n.nspname !~ '^pg_toast'
+AND c.relkind='i'
```
Comment (sweatybridge) [inline internal/inspect/queries.go:110]:
```suggestion
AND c.relkind = 'i'
```

This exclude clause is now redundant

File: internal/inspect/queries.go [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+pg_stat_user_tables
+ORDER BY seq_scan DESC`
+
+const TABLE_INDEX_SIZE_QUERY = `SELECT c.relname AS table,
+pg_size_pretty(pg_indexes_size(c.oid)) AS index_size
+FROM pg_class c
+LEFT JOIN pg_namespace n ON (n.oid = c.relnamespace)
+WHERE NOT n.nspname LIKE ANY($1)
+AND n.nspname !~ '^pg_toast'
+AND c.relkind='r'
```
Comment (sweatybridge) [inline internal/inspect/queries.go:208]:
```suggestion
AND c.relkind = 'r'
```

Ditto

File: internal/inspect/queries.go [hunk 7]
```diff
@@ -0,0 +1,299 @@
...
+ORDER BY
+n_live_tup DESC`
+
+const TABLE_SIZE_QUERY = `SELECT c.relname AS name,
+pg_size_pretty(pg_table_size(c.oid)) AS size
+FROM pg_class c
+LEFT JOIN pg_namespace n ON (n.oid = c.relnamespace)
+WHERE NOT n.nspname LIKE ANY($1)
+AND n.nspname !~ '^pg_toast'
+AND c.relkind='r'
```
Comment (sweatybridge) [inline internal/inspect/queries.go:225]:
```suggestion
AND c.relkind = 'r'
```

Ditto

File: internal/inspect/queries.go [hunk 8]
```diff
@@ -0,0 +1,299 @@
...
+ORDER BY pg_table_size(c.oid) DESC`
+
+const TOTAL_INDEX_SIZE_QUERY = `SELECT pg_size_pretty(sum(c.relpages::bigint*8192)::bigint) AS size
+FROM pg_class c
+LEFT JOIN pg_namespace n ON (n.oid = c.relnamespace)
+WHERE NOT n.nspname LIKE ANY($1)
+AND n.nspname !~ '^pg_toast'
+AND c.relkind='i'`
```
Comment (sweatybridge) [inline internal/inspect/queries.go:233]:
```suggestion
AND c.relkind = 'i'`
```

File: internal/inspect/queries.go [hunk 9]
```diff
@@ -0,0 +1,299 @@
...
+
+const TOTAL_TABLE_SIZE_QUERY = `SELECT c.relname AS name,
+pg_size_pretty(pg_total_relation_size(c.oid)) AS size
+FROM pg_class c
+LEFT JOIN pg_namespace n ON (n.oid = c.relnamespace)
+WHERE NOT n.nspname LIKE ANY($1)
+AND n.nspname !~ '^pg_toast'
+AND c.relkind='r'
```
Comment (sweatybridge) [inline internal/inspect/queries.go:241]:
```suggestion
AND c.relkind = 'r'
```

Comment (encima) [reply internal/inspect/queries.go]: want me to remove these or do you want to @sweatybridge?

Comment (sweatybridge) [reply internal/inspect/queries.go]: I just committed them. Do you mind running through all the commands again to confirm everything works?

**Review by sweatybridge — APPROVED** [review]

**Review by sweatybridge — APPROVED** [review]
I've verified all commands locally and fixed the result types where appropriate.

A new change I made is to include managed schemas like auth, storage, extensions, etc. when inspecting table sizes, ie. `table-records-count`, `table-sizes`, `total-table-sizes`. Because records in these schemas are mostly inserted by the user, it would be a good idea to surface their sizes. Schemas with `pg_*` prefix and `information_schema` are still excluded from these queries.

In the future, we can perhaps move queries.go to individual `*.sql` files for better syntax checks.
